addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,23 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+
+    def _message_post_after_hook(self, new_message, message_values):
+        self._set_expected_revenue()
+        return super()._message_post_after_hook(new_message, message_values)
+
+    def action_cancel(self):
+        res = super().action_cancel()
+        self._set_expected_revenue()
+        return res
```

Comment:
( I think we do not want any update on cancel :) )

Reply:
If we not call our method inside `action_cancel` it will break in that case suppose amount on some quote is set as the expected revenue and we cancel that quote revenue will not be updated based on the remaining quote.
Let me know your opinion.

Reply:
I think we can keep it simple, no need to handle action_cancel :)